Rubberduck.Parsing/VBA/ComponentParseTask.cs

```diff
@@ -0,0 +1,143 @@
+                var stopwatch = Stopwatch.StartNew();
+                ITokenStream stream;
+                var tree = ParseInternal(code, out stream);
+                stopwatch.Stop();
+                if (tree != null)
+                {
+                    Debug.Print("IParseTree for component '{0}' acquired in {1}ms (thread {2})", _component.Name, stopwatch.ElapsedMilliseconds, Thread.CurrentThread.ManagedThreadId);
+                }
+
+                token.ThrowIfCancellationRequested();
+
+                var attributes = _attributeParser.Parse(_component);
+
+                token.ThrowIfCancellationRequested();
+
+                ParseCompleted.Invoke(this, new ParseCompletionArgs
+                {
+                    ParseTree = tree,
+                    Tokens = stream,
+                    Attributes = attributes,
+                });
+            }
+            catch (COMException exception)
+            {
+                Debug.WriteLine("Exception thrown in thread {0}:\n{1}", Thread.CurrentThread.ManagedThreadId, exception);
+                ParseFailure.Invoke(this, new ParseFailureArgs
+                {
+                    Cause = exception
+                });
+            }
+            catch (SyntaxErrorException exception)
+            {
+                Debug.WriteLine("Exception thrown in thread {0}:\n{1}", Thread.CurrentThread.ManagedThreadId, exception);
+                ParseFailure.Invoke(this, new ParseFailureArgs
+                {
+                    Cause = exception
+                });
+            }
+            catch (OperationCanceledException cancel)
+            {
+                Debug.WriteLine("Operation was Cancelled", cancel);
+                // no results to be used, so no results "returned"
+                ParseCompleted.Invoke(this, new ParseCompletionArgs());
+            }
+        }
+
+        private string RewriteAndPreprocess()
+        {
+            var code = _rewriter == null ? string.Join(Environment.NewLine, _component.CodeModule.GetSanitizedCode()) : _rewriter.GetText();
+            string processed;
+            try
+            {
+                processed = _preprocessor.Execute(code);
+            }
+            catch (VBAPreprocessorException)
+            {
+                Debug.WriteLine("Falling back to no preprocessing");
+                processed = code;
+            }
+            return processed;
+        }
+
+        private static IParseTree ParseInternal(string code, out ITokenStream outStream)
+        {
+            var stream = new AntlrInputStream(code);
+            var lexer = new VBALexer(stream);
+            var tokens = new CommonTokenStream(lexer);
+            var parser = new VBAParser(tokens);
+
+            parser.AddErrorListener(new ExceptionErrorListener());
+
+            outStream = tokens;
+            return parser.startRule();
+        }
+
+        public void Parse()
+        {
+            ParseAsync(CancellationToken.None).Wait();
```

Comment:
It makes no sense for the ComponentParseTask to return a Task or to be structured Async in the first place. 

It should only provide a possibility for a Cancellation token, which may or may not be used by the RubberduckParser's Parse methods.  
That's exactly where the nature of the task is decided, the Task itself doesn't know, which is why this was removed